apps/extension/src/extension/Session.ts

```diff
@@ -0,0 +1,120 @@
+import browser, { Runtime } from "webextension-polyfill";
+import { LockKeyRingMsg } from "background/keyring";
+import { Ports } from "router";
+import { ExtensionRequester } from "./ExtensionRequester";
+
+export enum SessionMsgType {
+  Connect = "connect",
+  Disconnect = "disconnect",
+  Message = "message",
+}
+
+export type SessionMsg = {
+  msg: string;
+  type: SessionMsgType;
+};
+
+const SESSION_PORT = "session-port";
+// Maximum length of inactivity
+const SESSION_DURATION = 4 * 60 * 1000;
+
+type Timeout = ReturnType<typeof setTimeout>;
+// Connect to SESSION_PORT
+const initPort = (): Runtime.Port =>
+  browser.runtime.connect({
+    name: SESSION_PORT,
+  });
+
+const clearSessionTimeout = (timeout?: Timeout): void => {
+  if (timeout) {
+    clearTimeout(timeout);
+  }
+};
+
+/**
+ * Maintain an open port to the background until a certain limit has been reached.
+ */
+export class Session {
+  protected port: Runtime.Port | undefined;
+  private _requester: ExtensionRequester | undefined;
+  private _timestamp: number = Date.now();
+  private _timeout: Timeout | undefined;
+
+  public start(requester?: ExtensionRequester): Runtime.Port {
```

Comment:
nit: I feel that a short description of the task of this method would be especially helpful here.